src/main/java/com/comphenix/protocol/utility/MinecraftReflection.java

```diff
@@ -1579,4 +1576,27 @@ private static void useFallbackServer() {
 		setMinecraftClass("MinecraftServer", params[0]);
 		setMinecraftClass("PlayerList", params[1]);
 	}
+
+	public static Class<?> getLevelChunkPacketDataClass() {
+		return getNullableNMS("network.protocol.game.ClientboundLevelChunkPacketData");
+	}
+
+	public static Class<?> getLightUpdatePacketDataClass() {
+		return getNullableNMS("network.protocol.game.ClientboundLightUpdatePacketData");
+	}
+
+	public static Class<?> getBlockEntityTypeClass() {
+		return getMinecraftClass("world.level.block.entity.BlockEntityType", "world.level.block.entity.TileEntityTypes", "TileEntityTypes");
+	}
+
+	private static Class<?> blockEntityInfoClass;
+
+	public static Class<?> getBlockEntityInfoClass() {
+		if (blockEntityInfoClass == null) {

```

Comment:
Instead of caching you can do:
```
try {
  getMinecraftClass("BlockEntityInfo");
} catch (RuntimeException ex) { // should be runtime iirc
  Class<?> resolved = ...;
  setMinecraftClass("BlockEntityInfo", resolved);
  return resolved
```